SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -589,24 +603,30 @@ void UGlobalStateManager::QueryGSM(bool bRetryUntilAcceptingPlayers)
 		}
 		else
 		{
-			bool bNewAcceptingPlayers = GetAcceptingPlayersFromQueryResponse(Op);
+			bool bNewAcceptingPlayers = false;
+			int32 QuerySessionId = 0;
+			bool bQueryResponseSuccess = GetAcceptingPlayersAndSessionIdFromQueryResponse(Op, bNewAcceptingPlayers, QuerySessionId);
 
-			if (!bNewAcceptingPlayers && bRetryUntilAcceptingPlayers)
+			if (!bQueryResponseSuccess)
 			{
-				UE_LOG(LogGlobalStateManager, Log, TEXT("Not yet accepting new players. Will retry query for GSM."));
-				RetryQueryGSM(bRetryUntilAcceptingPlayers);
+				UE_LOG(LogGlobalStateManager, Error, TEXT("Failed to extract AcceptingPlayers and SessionId from GSM query response. Will retry query for GSM."));
+			}
+			else if (bNewAcceptingPlayers != bExpectedAcceptingPlayers ||
+					 QuerySessionId != ExpectedSessionId)
+			{
+				UE_LOG(LogGlobalStateManager, Log, TEXT("GlobalStateManager did not match expected query state. Will retry query for GSM."));
```

Comment:
Could you append the time expected to wait for the next query please.

Reply:
It's logged in a retry function. Hope that is alright.